src/main/ApplicationUtils.cpp

```diff
@@ -130,6 +182,35 @@ runWithConfig(Config cfg, std::optional<CatchupConfiguration> cc)
     return 0;
 }
 
+bool
+rebuildLatestState(Application& app)
+{
+    auto has = app.getLedgerManager().getLastClosedLedgerHAS();
+    auto lclHash =
+        app.getPersistentState().getState(PersistentState::kLastClosedLedger);
+
```

Comment:
Add a `releaseAssertOrThrow` here that the ledger tables are all empty (using `LedgerTxn::countObjects`). We never want this code to run on a nonempty ledger.

Reply:
this sounds like a useful check, but the problem is that calling `countObjects` is not allowed on non-root LedgerTxn, which is what we use in in-memory mode (that is, the never committing LedgerTxn). There doesn't seem to be any other machinery to verify LedgerTxn's state (we could add some, but I don't think we should at this time). I will add a comment about this though.

Reply:
I agree with Marta, I think it's fine as is for now